Rb 4.1.x nadenislamarre x86

nadenislamarre opened the pull request:
x86 first pr.

almost all is working correctly

Some things to continue to work on :
* es : keyboard works but not in textbox
* kodi : keyboard doesn't work if launched from es
* hardware : wiimote patch
* emulator : Scumm (compilation fails) (not analysed)
* emulator : reicast doesn't work
* emulator : zoom is not correct (too zoomed (sega) or not enought zoomed (gameboy))
* acpi : my power button doesnt ask the recalbox to shut down
* emulator : adjust resolution for performances (lower resolution / list of available resolutions)
* review the upgrade process (check it works)

Some things to know before merging :
- possible incompatibilities with noobs for rpi (i don't know noobs, and i changed the way the recalbox find its partition, see recalbox-part.sh)
- there is a dependance between x86 and x86_64 built : the configurable path ../recalbox-buildroot-x86_64/output/images/EFI.tar.gz is required so that 64 bit computer with efi can boot from the usb key.
I can provide this file externally.
- x86_64 port is already done too. but i wait that x86 is finnished to finnish it.
- please read each comment of each commit
- and don't hesitate if you've any question.

digitalLumberjack commented on board/recalbox/fsoverlay/etc/fstab:
```diff
@@ -1,12 +1,10 @@
 # /etc/fstab: static file system information.
 #
 # <file system> <mount pt>     <type>	<options>         <dump> <pass>
-/dev/root       /              ext4	ro,noauto         0      1
```
noobs sed those two variables.
https://github.com/recalbox/recalbox-rescue/blob/rb-4/sdcontent/os/recalboxOS-rpi3/partition_setup.sh#L16

Did not read the whole pr yet though

digitalLumberjack commented:
I'll keep reading tomorrow :) great job

substring commented:
Noobs for pc ? Grub, yeah !

digitalLumberjack commented:
read again :D
I said noobs sed (not set) it, so if it disapear, may cause issues (for multiboot, or even installation :)

nadenislamarre commented:
@substring note that we can do better than noobs for pc ;-)
if in a future we do efi + squashfs, install recalbox on a usb key is just an unzip on a fat32 key => no need of boot and recalbox partition, only 1. this is not for now.

digitalLumberjack commented on board/recalbox/fsoverlay/etc/init.d/S02splash:
```diff
@@ -51,7 +51,15 @@ do_start ()
-    fbv -f -i /recalbox/system/resources/splash/logo-version.png
+
+    # on some sytems, fb0 is not immediatly loaded, so, keep a chance by waiting a bit
+    N=0
```
N=15

digitalLumberjack replied:
So we can visualize the number of try

digitalLumberjack commented on board/recalbox/fsoverlay/etc/init.d/S02splash:
```diff
@@ -51,7 +51,15 @@ do_start ()
+
+    # on some sytems, fb0 is not immediatly loaded, so, keep a chance by waiting a bit
+    N=0
+    while ! test -e /dev/fb0 -o $N -gt 15
```
-lt 0

digitalLumberjack commented on board/recalbox/fsoverlay/etc/init.d/S02splash:
```diff
@@ -51,7 +51,15 @@ do_start ()
+    while ! test -e /dev/fb0 -o $N -gt 15
+    do
+	sleep 1
+	let N++
```
N--

digitalLumberjack replied:
recalbox-part.sh replace the fstab, need to change the sed in https://github.com/recalbox/recalbox-rescue/blob/rb-4/sdcontent/os/recalboxOS-rpi3/partition_setup.sh before merging (i'll do)

digitalLumberjack commented:
recalbox-part.sh replace the fstab, need to change the sed in https://github.com/recalbox/recalbox-rescue/blob/rb-4/sdcontent/os/recalboxOS-rpi3/partition_setup.sh before merging (i'll do)

digitalLumberjack commented on board/recalbox/fsoverlay/recalbox/scripts/recalbox-config.sh:
```diff
@@ -251,21 +251,33 @@ fi
 if [ "$command" == "audio" ];then
+    # this code is specific to the rpi
+    # don't set it on other boards
+    # find a more generic way would be nice
```
alsamixer ?

digitalLumberjack commented on board/recalbox/recalbox-patch-target.sh:
```diff
@@ -22,6 +22,18 @@ ln -sf "/recalbox/share/cheats"                                       "${TARGET_
+fi
+
+# we don't want default xorg files
+rm -f "${TARGET_DIR}/etc/X11/xorg.conf" || exit 1
```
rm -f never return != 0

digitalLumberjack replied:
whatever

digitalLumberjack commented on package/busybox/busybox.config:
```diff
@@ -25,7 +25,7 @@ CONFIG_FEATURE_VERBOSE_USAGE=y
 # CONFIG_FEATURE_COMPRESS_USAGE is not set
 CONFIG_FEATURE_INSTALLER=y
 # CONFIG_INSTALL_NO_USR is not set
-CONFIG_LOCALE_SUPPORT=y
```
We need that for special chars in bash no ?

digitalLumberjack commented:
Review finished. Great job.
We discuss about the fex things i commented, we change the recalbox-rescue scripts and go for merge :)

nadenislamarre commented:
-rm -f /tmp/nae || echo ahbon
rm: ne peut enlever `/tmp/nae': Opération non permise

## ahbon

nadenislamarre commented:
> > We need that for special chars in bash no ?
> > when i did the busybox config the first time some monthes ago, i modified the bad file.
> > i just moved my busybox config to the custom file done for this.

digitalLumberjack commented:
`rm -f /tmp/nae; echo $?` 
0
Seems to be shell dependent

nadenislamarre commented:
no, it's an answer from rm.
note that i switch of user to get this error.

the only difference between rm and rm -f
is that
1- if the file doesn't exist, rm -f returns 0
2- no prompt

so, i case you've not the right to remove a file, both are failing for
example.

digitalLumberjack commented:
okay :)

digitalLumberjack commented:
Let's do that.

digitalLumberjack commented:
@ian57 didn't see your message. Is it still valid ?

digitalLumberjack commented:
Hum
this is what we aim : https://github.com/libretro/picodrive/blob/master/Makefile.libretro#L324

ian57 commented:
No your modification is ok... did not see the arm part! sorry
